### PR TITLE
Support new test file conventions

### PR DESCRIPTION
**Pull request by kentcdodds**
https://github.com/sindresorhus/ava/pull/721

### CODE REVIEW

File: test/no-ignored-test-files.js [hunk 1]
```diff
@@ -79,20 +83,20 @@ test('without AVA config in package.json', () => {
 			},
 			{
 				code: code(true),
-				filename: toPath('lib/foo.test.js'),
-				errors: [{message: 'Test file is ignored because it is not in `test.js test-*.js test/**/*.js`'}]
+				filename: toPath('lib/foo.spec.js'),
+				errors: [{message: 'Test file is ignored because it is not in `test.js test-*.js test/**/*.js **/__tests__/** **/*.test.js`'}]
```
Comment (jfmengels) [inline test/no-ignored-test-files.js]: `**/__tests__/**` --> `**/__tests__`

Comment (kentcdodds) [inline test/no-ignored-test-files.js]: You didn't see anything....

penguin-oops-XItRQJP0wai7m

**Comment by jfmengels** [comment]
Let's wait until sindresorhus/ava#721 gets merged, but LGTM at the moment :)
Thanks!

File: rules/no-ignored-test-files.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ var createAvaRule = require('../create-ava-rule');
 var defaultFiles = [
 	'test.js',
 	'test-*.js',
-	'test/**/*.js'
+	'test/**/*.js',
+	'**/__tests__',
```
Comment (sindresorhus) [inline rules/no-ignored-test-files.js]: Should be `**/__tests__/**/*.js` as this plugin doesn't have directory recursion like AVA.

**Comment by forresst** [comment]
@kentcdodds We must also add in the documentation [no-ignored-test-files.md](https://github.com/sindresorhus/eslint-plugin-ava/blob/master/docs/rules/no-ignored-test-files.md), replace:
```markdown
By default, it will search in `test.js test-*.js test/**/*.js`, which you can override by specifying a path when launching AVA or in the [AVA configuration in the `package.json` file](https://github.com/sindresorhus/ava#configuration).
```
By:
```markdown
By default, it will search in `test.js test-*.js test/**/*.js  **/__tests__/**/*.js **/*.test.js`, which you can override by specifying a path when launching AVA or in the [AVA configuration in the `package.json` file](https://github.com/sindresorhus/ava#configuration).
```

**Comment by kentcdodds** [comment]
This PR has been updated based on feedback :-)

**Comment by sindresorhus** [comment]
Thanks @kentcdodds 🎉

**Comment by kentcdodds** [comment]
funny-pokemon-laughing-Jir3toQTWW9Ne

**Comment by jfmengels** [comment]
Thanks a lot!

**Pull request merged**